src/Azure/Orleans.Clustering.AzureStorage/OrleansSiloInstanceManager.cs

```diff
@@ -25,38 +25,44 @@ internal class OrleansSiloInstanceManager
         private readonly AzureTableDataManager<SiloInstanceTableEntry> storage;
         private readonly ILogger logger;
 
-        internal static TimeSpan initTimeout = AzureTableDefaultPolicies.TableCreationTimeout;
```

Comment:
Shouldn't we get rid of AzureTableDefaultPolicies.TableCreationTimeout, and AzureTableDefaultPolicies.TableOperationTimeout.  It would be best to be consistent, and not have duplicate settings.

Reply:
It's not really duplicate, since users have the option to not specify a timeout, in which case Orleans would need to default to a value.

Reply:
My main concern is that if we've duplicate settings for the same behavior it will be more difficult for us and other users to figure out which they should be setting/using.  As it is, we've a single consistent place for these values, but no good pattern for users to specify them, especially on a per provider basis.  If we're going to address this, my preference would be to do so consistently.  Even if we don't change all of the azure settings in this PR, the pattern should be consistent and extendable to all azure storage points.

As you probably only need this for specific areas, I don't think it's reasonable to block this change due to incompleteness, when we can create a work item to migrate the changes across the other storage points, but it is important to get the pattern consistent.

I noted this issue a while back, but it's been in the backlog for a bit.  https://github.com/dotnet/orleans/issues/4557  @ReubenBond, @sergeybykov, @benjaminpetit, thoughts?  If I'm the only one concerned, then I'm probably being too nit-picky (it happens :])

Reply:
Yes, I think we should remove `AzureTableDefaultPolicies` altogether and adopt a consistent pattern.

Regarding this inheritance-based design, we used a similar pattern for {Client,Silo}MessagingOptions, and regretted it, because users made the mistake of configuring the base class (`.Configure<AzureStorageOperationOptions>(...)`) instead of the correct class, resulting in misconfiguration (no configuration). See #5574, for example.

I would prefer we use composition instead of inheritance. For example, by adding a field called `StoragePolicy`, of type `AzureStoragePolicyOptions` to the existing provider options (`AzureStorageClusteringOptions`, etc). That policy class would contain the properties currently in `AzureTableDefaultPolicies`.

How does that sound?

Reply:
Regarding inheritance vs composition, the inheritance pattern used in this PR 'shouldn't' have the same issue we ran into with MessagingOptions because the base class is abstract, so it's comparable to an interface, hence I was less concerned with that.

For consistency, I think we should settle on what behaviors we want to support.  I'm confident we minimally want to allow users to provide their own policy settings per provider.  Do we want to also support a configurable global default?

If we don't want a configurable global default, then an inheritance model or composition model with default values both 'should' get us what we want.  If we want to support a configurable global default, things get more complicated, because we'd need a way to fallback to a configured global default if settings were not configured in the provider (as we do with ClusterOptions).  For this reason, I suspect supporting a configurable global default is probably not warranted.

If we agree that a configurable global default isn't needed, then we need only move the defaults from AzureTableDefaultPolicies (and remove that class) to the new configuration class (whether it's introduced via inherited or composed).

Reply:
But the thing is that this "StoragePolicy" property has to be applied to all AzureStorage<>Options anyway which requires some form of inheritance to avoid duplication.

Reply:
Spoke with Reuben offline, tableName is now abstract, and the policyOptions is now injectable into every Orleans Provider. Removed the Global config.